src/Lecture2.hs

```diff
@@ -74,7 +80,11 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt _ [] = (Nothing, [])
+removeAt x arr
+  | x < 0 = (Nothing, arr)
+  | otherwise = let dropped = drop x arr in (if not (null dropped) then Just (head dropped) else Nothing, take x arr ++ drop (x + 1) arr)
```

Comment:
That's a correct implementation 👍🏻 
Alternatively, you can use a standard function [splitAt](https://hackage.haskell.org/package/base-4.16.0.0/docs/Prelude.html#v:splitAt) instead of using `take` and `drop` separately and avoid traversing the list prefix twice 🔃